### PR TITLE
Add DataPersistenceConfig to CacheConfigHolder and AddCacheConfigMessageTask

### PR DESCRIPTION
**Pull request by ufukyilmaz**
Also, this PR adds null checks for fields that are incorrectly marked
as nullable. 
Continuation of https://github.com/hazelcast/hazelcast/pull/21432

Client protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/425
Fixes https://github.com/hazelcast/hazelcast-enterprise/issues/4282

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by mdumandag — APPROVED** [review]
LGTM

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java [hunk 1]
```diff
@@ -86,9 +95,10 @@ protected IdentifiedDataSerializable getConfig() {
         config.setValueType(parameters.valueType);
         config.setWanReplicationRef(parameters.wanReplicationRef);
         config.setWriteThrough(parameters.writeThrough);
-        if (parameters.isMerkleTreeConfigExists) {
+        if (parameters.isMerkleTreeConfigExists && parameters.merkleTreeConfig != null) {
             config.setMerkleTreeConfig(parameters.merkleTreeConfig);
         }
+        config.setDataPersistenceConfig(parameters.dataPersistenceConfig);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java:101]: Should be conditionally set if `parameters.dataPersistenceConfig != null`. Consider the case of an older (eg 5.1) client sending an `AddCacheConfig` message: `dataPersistenceConfig` will be `null` and `config.setDataPersistenceConfig` will throw `NullPointerException`. See similar issue reported in #21732 for map config.

Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java]: Thanks for catching it. I'll take care of the add map config too.

**Review by vbekiaris — APPROVED** [review]

**Comment by ufukyilmaz** [comment]
Hi @hazelcast/apisjava,
Could you handle the merge of this PR when it's possible? I believe it's ready to merge. 

Note that: this is not an urgent PR. I just hesitated to merge it on my own to avoid making the client protocol PR side out-of-sync.

**Pull request merged**